File: examples/dotnet/Examples/Business.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using MongoDB.Bson;
using Realms;
using Realms.Sync;

namespace Examples
{
    public class Business : RealmObject
    {
        [PrimaryKey]
        [MapTo("_id")]
        public ObjectId Id { get; set; } = ObjectId.GenerateNewId();

        public string Name { get; set; }
        public IList<Address> addresses { get; }
    }
}

File: examples/react-native/__tests__/js/Models/Business.js
import Realm from 'realm';

// :snippet-start: js-business-schema
class Business extends Realm.Object {
  static schema = {
    name: "Business",
    primaryKey: "_id",
    properties: {
      _id: "objectId",
      name: "string",
      addresses: { type: "list?", objectType: "Address" }, // Embed an array of objects
    },
  };
}
// :snippet-end:
export default Business;
